### PR TITLE
Adding Zodiark Split keyboard

### PR DESCRIPTION
**Pull request by Aleblazer**
## Description

Adding Aleblazer/Zodiark split keyboard and keymaps

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/aleblazer/zodiark/zodiark.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+#include "quantum.h"
+#include "split_util.h"
+
+#define LAYOUT( \
+ L00, L01, L02, L03, L04, L05, L06, R00, R01, R02, R03, R04, R05, R06, \
```
Comment (zvecr) [inline keyboards/aleblazer/zodiark/zodiark.h:22]: Top part of layout macro should represent physical layout, currently it does not match the image provided in the readme

File: keyboards/aleblazer/zodiark/rules.mk [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+NKRO_ENABLE ?= no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
```
Comment (zvecr) [inline keyboards/aleblazer/zodiark/rules.mk:63]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
RGBLIGHT_ENABLE = no
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812
```

File: keyboards/aleblazer/zodiark/config.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (zvecr) [inline keyboards/aleblazer/zodiark/config.h:18]:
```suggestion
#pragma once
```

File: keyboards/aleblazer/zodiark/config.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xF901
+#define PRODUCT_ID      0xF902
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Aleblazer
+#define PRODUCT         Zodiark
+#define DESCRIPTION     68 key split keyboard
+
```
Comment (zvecr) [inline keyboards/aleblazer/zodiark/config.h:29]:
```suggestion

```

File: keyboards/aleblazer/zodiark/config.h [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 150 // limits maximum brightness of LEDs to 150 out of 255. Higher may cause the controller to crash.
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_HUE_STEP 8
+#define RGB_MATRIX_SAT_STEP 8
+#define RGB_MATRIX_VAL_STEP 8
+#define RGB_MATRIX_SPD_STEP 10
+#endif
+
+#endif
```
Comment (zvecr) [inline keyboards/aleblazer/zodiark/config.h:86]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by Aleblazer** [comment]
Applied all change requests, and thanks to ridingqwerty in the QMK Discord, fixed the zodiark.h layout and all the keymaps to reflect the shift in key locations.

File: keyboards/aleblazer/zodiark/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+```sh
+make zodiark:default
+```
+
+Flash the default keymap:
+
+```sh
+make zodiark:default:avrdude
+```
```
Comment (fauxpark) [inline keyboards/aleblazer/zodiark/readme.md:21]:
```suggestion
* Keyboard Maintainer: [Aleblazer](https://github.com/Aleblazer/), [Discord Link](https://discord.gg/BCSbXwskVt)
* Hardware Supported: Pro Micro and derivatives
* Hardware Availability: Pending Group Buy - [Discord Link](https://discord.gg/BCSbXwskVt)

Make example for this keyboard (after setting up your build environment):

    make zodiark:default

Flashing example for this keyboard:

    make zodiark:default:flash

File: keyboards/aleblazer/zodiark/zodiark.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+#include "quantum.h"
+#include "split_util.h"
```
Comment (fauxpark) [inline keyboards/aleblazer/zodiark/zodiark.h:19]:
```suggestion
```

File: keyboards/aleblazer/zodiark/rules.mk [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
```
Comment (fauxpark) [inline keyboards/aleblazer/zodiark/rules.mk:27]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Aleblazer** [comment]
Applied all suggestions from fauxpark, and added RGB_MATRIX_DRIVER = WS2812 to my rgbmatrixwip keymap rules.mk.

File: keyboards/aleblazer/zodiark/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/aleblazer/zodiark/config.h [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+#define VENDOR_ID       0xF901
+#define PRODUCT_ID      0xF902
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Aleblazer
+#define PRODUCT         Zodiark
+#define MASTER_LEFT
+
+#ifdef OLED_DRIVER_ENABLE
+  #define OLED_TIMEOUT 400000
+#endif
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/config.h:31]: Is this is defined at the keyboard level, it probably doesn't need to be fined here.
```suggestion
```

File: keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/keymap.c [hunk 11]
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/rules.mk [hunk 12]
```diff
@@ -0,0 +1,2 @@
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/rules.mk:2]: This should be moved to the keyboard's config.

File: keyboards/aleblazer/zodiark/keymaps/slimoled/keymap.c [hunk 13]
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/keymaps/slimoled/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Aleblazer** [comment]
Cleaned up the OLED code from the keyboard level config.h, added the GPL2+ license header to all keymaps, and removed the rgbmatrixwip keymap (Matrix split effects didn't work without a custom solution at this time).

File: keyboards/aleblazer/zodiark/config.h [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xF901
+#define PRODUCT_ID      0xF902
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Aleblazer
+#define PRODUCT         Zodiark
+#define MASTER_LEFT
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/config.h:27]: This isn't actually needed, as this is the default behavior. 
```suggestion
```

File: keyboards/aleblazer/zodiark/config.h [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+#define NO_ACTION_MACRO
+
+#define USE_SERIAL
```
Comment (drashna) [inline keyboards/aleblazer/zodiark/config.h:31]: This is the default behavior, so this isn't needed.
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/aleblazer/zodiark/readme.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
+# Zodiark Keyboard
+
+![Zodiark Split](https://i.imgur.com/49O8aow.jpg)
```
Comment (fauxpark) [inline keyboards/aleblazer/zodiark/readme.md:3]:
```suggestion
![Zodiark Split](https://i.imgur.com/49O8aowl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Aleblazer** [comment]
All requested changes have been made, and added a VIA keymap.

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks"

**Pull request merged**